### PR TITLE
Import the code from the self provisioning plugin

### PR DESCRIPTION
**Pull request by lavocatt**
Please check the README to se what is supported.

### CODE REVIEW

**Comment by lavocatt** [comment]
@howardgao the tests are failing because the testsuite can't load the `.env` file do you think some configuration is missing on the github project?

edit: the file was ignored. I've added it back in the repo.

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env sh
+
+DEFAULT_IMAGE="quay.io/artemiscloud/activemq-artemis-jolokia-api-server:latest"
+PLUGIN_IMAGE=${DEFAULT_IMAGE}
```
Comment (gaohoward) [inline deploy.sh]: As the api-server is a standalone module, probably rename the PLUGIN_IMAGE to API_SERVER_IMAGE?

File: README.md [hunk 2]
```diff
@@ -1 +1,61 @@
...
+
+```
+yarn run build-api-doc
+```
+
+## Production build
+
+1. Build the image:
+   ```sh
+   docker build -t quay.io/artemiscloud/activemq-artemis-self-provisioning-plugin:latest .
```
Comment (gaohoward) [inline README.md]: the image name may be change to reflect api-server?

File: README.md [hunk 3]
```diff
@@ -1 +1,61 @@
...
+   ```
+2. Push the image to image registry:
+   ```sh
+   docker push quay.io/artemiscloud/activemq-artemis-self-provisioning-plugin:latest
```
Comment (gaohoward) [inline README.md]: same above. The image name shouldn't contain self-provisioning plugin

File: package.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    "pretty-quick": "3.1.3",
+    "sort-package-json": "^2.10.0",
+    "swagger-routes-express": "^3.3.2",
+    "ts-jest": "^29.2.1",
+    "ts-loader": "^9.3.1",
+    "ts-node": "10.9.2",
+    "typescript": "^4.7.4",
+    "yaml": "^2.4.5"
+  },
+  "consolePlugin": {
```
Comment (gaohoward) [inline package.json]: do we need this `consolePlugin`?

**Review by gaohoward — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (lavocatt) [reply package.json]: No we don't, thanks for catching these leftovers

File: deploy.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      printUsage
+      exit 1
+      ;;
+    *)
+      ;;
+  esac
+done
+
+echo "deploying using image: ${PLUGIN_IMAGE}"
+oc kustomize deploy | sed "s|image: .*|image: ${PLUGIN_IMAGE}|" | oc apply -f -
```
Comment (Msarawan) [inline deploy.sh]: Here also updating `PLUGIN_IMAGE` to `API_SERVER_IMAGE`

File: deploy.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -h|--help)
+      printUsage
+      exit 0
+      ;;
+    -i|--image)
+      PLUGIN_IMAGE="$2"
```
Comment (Msarawan) [inline deploy.sh]: same here updating `PLUGIN_IMAGE` with `API_SERVER_IMAGE`

**Review by Msarawan — APPROVED** [review]
LGTM. nice work Thomas!

**Review by gaohoward — APPROVED** [review]

**Pull request merged**